rasa/core/training/loading.py

```diff
@@ -0,0 +1,116 @@
+import logging
+import os
+from pathlib import Path
+from typing import Text, Optional, Dict, List, Union
+
+import rasa.utils.io as io_utils
+from rasa.core.domain import Domain
+from rasa.core.interpreter import NaturalLanguageInterpreter, RegexInterpreter
+from rasa.core.training.story_reader.story_reader import StoryReader
+from rasa.core.training.structures import StoryStep
+from rasa.data import YAML_FILE_EXTENSIONS, MARKDOWN_FILE_EXTENSION
+
+logger = logging.getLogger(__name__)
+
+
+def _get_reader(
+    filename: Text,
+    domain: Domain,
+    interpreter: NaturalLanguageInterpreter = RegexInterpreter(),
+    template_variables: Optional[Dict] = None,
+    use_e2e: bool = False,
+) -> StoryReader:
+    from rasa.core.training.story_reader import YAMLStoryReader
+    from rasa.core.training.story_reader import MarkdownStoryReader
+
+    if filename.endswith(MARKDOWN_FILE_EXTENSION):
+        return MarkdownStoryReader(
+            interpreter, domain, template_variables, use_e2e, filename
+        )
+    elif filename.split(".")[-1] in YAML_FILE_EXTENSIONS:
+        return YAMLStoryReader(
+            interpreter, domain, template_variables, use_e2e, filename
+        )
+
+    raise Exception(
```

Comment:
Can we raise some subclass of `Exception`? Like a `ValueError` or we define a `TrainingDataImportErrror`?